tests/core/test_core_edm_host.cpp

```diff
@@ -213,3 +212,91 @@ TEST_F(core_edm_host_test, const_proxy) {
         }
     }
 }
+
+namespace {
+using simple_schema = vecmem::edm::schema<vecmem::edm::type::scalar<int>,
+                                          vecmem::edm::type::vector<float>>;
+template <typename BASE>
+struct simple_interface : public BASE {
+    using BASE::BASE;
+    using BASE::operator=;
+    simple_interface(int scalar, float vector) : BASE{{scalar, vector}} {}
+    auto& scalar() { return BASE::template get<0>(); }
+    const auto& scalar() const { return BASE::template get<0>(); }
+    auto& vector() { return BASE::template get<1>(); }
+    const auto& vector() const { return BASE::template get<1>(); }
+};
+using simple_host_type =
+    simple_interface<vecmem::edm::host<simple_schema, simple_interface>>;
+}  // namespace
+
+TEST_F(core_edm_host_test, push_back_simple) {
+
+    // Set up a simple type, without any jagged vectors.
+    simple_host_type host{m_resource};
+
+    // Add a couple of elements to it. Since simple_interface has an explicit
+    // constructor, the formalism here is fairly simple.
+    host.push_back({1, 2.f});
+    host.push_back({2, 3.f});
+
+    // Check that the contents are as expected.
+    ASSERT_EQ(host.size(), 2);
+    EXPECT_EQ(host.at(0).scalar(), 2);
+    EXPECT_EQ(host.at(1).scalar(), 2);
+    EXPECT_FLOAT_EQ(host.at(0).vector(), 2.f);
+    EXPECT_FLOAT_EQ(host.at(1).vector(), 3.f);
+}
+
+TEST_F(core_edm_host_test, push_back_jagged) {
+
+    // Create an empty host container.
+    host_type host{m_resource};
+
+    // Add a couple of elements to it. Since interface has no constructor of
+    // its own, because with jagged vectors that's not so easy to do, one needs
+    // to use double braces in the following expression.
+    host.push_back({{1, 2.f, vecmem::vector<double>{3., 4., 5.}}});
+    host.push_back({{2, 3.f, vecmem::vector<double>{4., 5., 6.}}});
```

Comment:
I'm not absolutely happy about the formalism here, but I couldn't easily come up with one that would not require double curly braces. 🤔

Reply:
Turns out that it was not all that hard to implement it in a way that results in a more natural formalism for the client code. 🤔 (All the infrastructure was already in place to make it happen.)